### DISCUSSION TITLE
how to stop checking specific header files

### DISCUSSION THREAD

**Discussion by a-Void-Deer** [clangd/clangd#2164] [Q&A] [answered]
My clangd keeps checking header files like <windows.h> and shows me errors. 
![image](https://github.com/user-attachments/assets/7f6ab4cb-1002-4c95-8d37-ca67dd7c7c9b)

I can't find config.yaml in the directory (there are nothing named as "slangd" or "config.yaml" in %LocalAppData%), so I created one.
![image](https://github.com/user-attachments/assets/f84b5b4f-536b-427d-8c6e-21a57a50ec01)
I've tried both of these, but they didn't work. What should I do now?

**Comment by HighCommander4** [comment, marked as answer]
`IgnoreHeaders` only applies to [include-cleaner](https://clangd.llvm.org/design/include-cleaner) diagnostics.

`PathMatch` is evaluated againt the path of the source file open in the editor, not the path of the file in which the diagnostic originates.

I'm not aware of any mechanism for ignoring all diagnostics originating from a specific header file. For best results I recommend investigating and resolving the underlying cause of the error, not trying to ignore it.

**Reply by a-Void-Deer** [reply]
I checked the header file, it's <stdlib.h> which has a lot of errors. It's provided by minGW-w64. 
I'm a beginner, but I'll try quick fix anyway. Thank you!
I'll upload a zip file containing the stdlib.h.
[stdlib.zip](https://github.com/user-attachments/files/17158130/stdlib.zip)

**Reply by HighCommander4** [reply]
If you'd like help figuring out the errors, please post [clangd logs](https://clangd.llvm.org/troubleshooting.html#gathering-logs).

**Reply by a-Void-Deer** [reply]
[clangd.log](https://github.com/user-attachments/files/17163910/clangd.log)
Did you mean this?

Also, I've tried quick fix in that header files, and I haven't met anything unexpected yet, so I'll mark the question as answered. Thank you 
 again :)
